fmriprep/workflows/bold/base.py

```diff
@@ -603,45 +603,51 @@ def init_func_preproc_wf(bold_file, ignore, freesurfer,
+        ica_aroma_wf = init_ica_aroma_wf(
+            template=template,
+            mem_gb=mem_gb['resampled'],
+            omp_nthreads=omp_nthreads,
+            use_compression=not (low_mem and use_aroma),
+            use_fieldwarp=fmaps is not None,
+            ignore_aroma_err=ignore_aroma_err,
+            name='ica_aroma_wf')
+        join = pe.Node(JoinTSVColumns(), name='aroma_confounds')
 
-            workflow.disconnect([
-                (bold_confounds_wf, outputnode, [
-                    ('outputnode.confounds_file', 'confounds'),
-                ]),
-            ])
-            workflow.connect([
-                (bold_hmc_wf, ica_aroma_wf, [
-                    ('outputnode.movpar_file', 'inputnode.movpar_file')]),
-                (bold_mni_trans_wf, ica_aroma_wf, [
-                    ('outputnode.bold_mask_mni', 'inputnode.bold_mask_mni'),
-                    ('outputnode.bold_mni', 'inputnode.bold_mni')]),
-                (bold_confounds_wf, join, [
-                    ('outputnode.confounds_file', 'in_file')]),
-                (ica_aroma_wf, join,
-                    [('outputnode.aroma_confounds', 'join_file')]),
-                (ica_aroma_wf, outputnode,
-                    [('outputnode.aroma_noise_ics', 'aroma_noise_ics'),
-                     ('outputnode.melodic_mix', 'melodic_mix'),
-                     ('outputnode.nonaggr_denoised_file', 'nonaggr_denoised_file')]),
-                (join, outputnode, [('out_file', 'confounds')]),
-            ])
+        workflow.disconnect([
+            (bold_confounds_wf, outputnode, [
+                ('outputnode.confounds_file', 'confounds'),
+            ]),
+        ])
+        workflow.connect([
+            (inputnode, ica_aroma_wf, [
+                ('bold_file', 'inputnode.name_source'),
+                ('t1_2_mni_forward_transform', 'inputnode.t1_2_mni_forward_transform')]),
+            (bold_split, ica_aroma_wf, [
+                ('out_files', 'inputnode.bold_split')]),
+            (bold_hmc_wf, ica_aroma_wf, [
+                ('outputnode.movpar_file', 'inputnode.movpar_file'),
+                ('outputnode.xforms', 'inputnode.hmc_xforms')]),
+            (bold_reg_wf, ica_aroma_wf, [
+                ('outputnode.itk_bold_to_t1', 'inputnode.itk_bold_to_t1')]),
+            (bold_bold_trans_wf, ica_aroma_wf, [
+                ('outputnode.bold_mask', 'inputnode.bold_mask')]),
+            (bold_sdc_wf, ica_aroma_wf, [
+                ('outputnode.out_warp', 'inputnode.fieldwarp')]),
+            (bold_confounds_wf, join, [
+                ('outputnode.confounds_file', 'in_file')]),
+            (ica_aroma_wf, join,
+                [('outputnode.aroma_confounds', 'join_file')]),
+            (ica_aroma_wf, outputnode,
+                [('outputnode.aroma_noise_ics', 'aroma_noise_ics'),
+                 ('outputnode.melodic_mix', 'melodic_mix'),
+                 ('outputnode.nonaggr_denoised_file', 'nonaggr_denoised_file')]),
```

Comment:
If we eventually support denoising in multiple spaces, we may want to give the non-aggressively denoised file a more specific name, such as `nonaggr_denoised_mni`, but this change is not directly related to this pull request.